refactor: Signet fixups

maflcko opened the pull request:
Some doc and test fixups for #18267

Commit:
doc: Document signet BIP

Commit:
fuzz: Remove needless guard

Commit:
test: Run signet test even when wallet was not compiled

Commit:
doc: Update comments for new chain settings (-signet and -chain=signet)

ajtowns commented on doc/bips.md:
```diff
@@ -42,4 +42,5 @@ BIPs that are implemented by Bitcoin Core (up-to-date up to **v0.21.0**):
 * [`BIP 173`](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki): Bech32 addresses for native Segregated Witness outputs are supported as of **v0.16.0** ([PR 11167](https://github.com/bitcoin/bitcoin/pull/11167)). Bech32 addresses are generated by default as of **v0.20.0** ([PR 16884](https://github.com/bitcoin/bitcoin/pull/16884)).
 * [`BIP 174`](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki): RPCs to operate on Partially Signed Bitcoin Transactions (PSBT) are present as of **v0.17.0** ([PR 13557](https://github.com/bitcoin/bitcoin/pull/13557)).
 * [`BIP 176`](https://github.com/bitcoin/bips/blob/master/bip-0176.mediawiki): Bits Denomination [QT only] is supported as of **v0.16.0** ([PR 12035](https://github.com/bitcoin/bitcoin/pull/12035)).
+* [`BIP 325`](https://github.com/bitcoin/bips/blob/master/bip-0325.mediawiki): Signet test network is supported as of **v0.21.0** ([PR 18267](https://github.com/bitcoin/bitcoin/pull/18267)).
```
(Not strictly true until bitcoin/bips#978 and bitcoin/bips#1000 are merged)

MarcoFalke replied:
heh, I hope that those are merged before we release the next major version ;)

kallewoof replied:
I've ACK'd both, so they should be considered merged, I think, since i'm the BIP author.

ajtowns commented on doc/release-notes.md:
```diff
@@ -335,6 +335,10 @@ RPC
 
+- The BIP 325 default signet can be enabled by the `-chain=signet` or `-signet`
+  setting. The settings `-signetchallenge` and `-signetseednode` allow
+  modifying the network.
```
"modifying the network" -> "enabling a custom signet instead" ?

MarcoFalke replied:
thx done

Commit:
refactor: Remove SignetTxs::m_valid and use optional instead

m_valid implies the block solution has been checked, which is not the
case. It only means the txs could be parsed. C++17 comes with
std::optional, so just use that instead.

ajtowns commented on src/signet.cpp:
```diff
@@ -65,8 +65,9 @@ static uint256 ComputeModifiedMerkleRoot(const CMutableTransaction& cb, const CB
-SignetTxs SignetTxs::Create(const CBlock& block, const CScript& challenge)
+Optional<SignetTxs> SignetTxs::Create(const CBlock& block, const CScript& challenge)
 {
+    const auto invalid = []() -> Optional<SignetTxs> { return nullopt; };
```
Should just replace `return invalid()` with `return nullopt` directly.

MarcoFalke replied:
thx done

ajtowns commented on src/signet.h:
```diff
@@ -22,21 +24,14 @@ bool CheckSignetBlockSolution(const CBlock& block, const Consensus::Params& cons
 
 public:
-    SignetTxs(const CBlock& block, const CScript& challenge) : SignetTxs(Create(block, challenge)) { }
+    static Optional<SignetTxs> Create(const CBlock& block, const CScript& challenge);
```
This is much better!

ajtowns reviewed (commented):
Concept ACK

fjahr commented:
@ajtowns suggestions are good, ACK otherwise.

fjahr commented:
Code review ACK fadee171c8b0b591daf8f1d65640e4f288b3c3bf

dr-orlovsky reviewed (approved):
Reviewed & ACK https://github.com/bitcoin/bitcoin/pull/19993/commits/facaf9e61f4b9ea91fab554d495ebea1043d08fb

ajtowns commented:
ACK facaf9e61f4b9ea91fab554d495ebea1043d08fb -- code review only

kallewoof reviewed (commented):
Code review ACK facaf9e61f4b9ea91fab554d495ebea1043d08fb
